src/framework/components/layout-group/layout-calculator.js

```diff
@@ -0,0 +1,628 @@
+                        );
+                    } else {
+                        element.entity.setLocalPosition(
+                            positionsThisLine[elementIndex][b.axis],
+                            positionsThisLine[elementIndex][a.axis],
+                            element.entity.getLocalPosition().z
+                        );
+                    }
+                }
+            }
+        }
+
+        // Reads all size-related properties for each element and applies some basic
+        // sanitization to ensure that minWidth is greater than 0, maxWidth is greater
+        // than minWidth, etc.
+        function getElementSizeProperties(elements) {
+            var sizeProperties = [];
+
+            for (var i = 0; i < elements.length; ++i) {
+                var element = elements[i];
+                var minWidth  = Math.max(getProperty(element, 'minWidth'), 0);
+                var minHeight = Math.max(getProperty(element, 'minHeight'), 0);
+                var maxWidth  = Math.max(getProperty(element, 'maxWidth'), minWidth);
+                var maxHeight = Math.max(getProperty(element, 'maxHeight'), minHeight);
+                var width  = clamp(getProperty(element, 'width'), minWidth, maxWidth);
+                var height = clamp(getProperty(element, 'height'), minHeight, maxHeight);
+                var fitWidthProportion  = getProperty(element, 'fitWidthProportion');
+                var fitHeightProportion = getProperty(element, 'fitHeightProportion');
+
+                sizeProperties.push({
+                    minWidth: minWidth,
+                    minHeight: minHeight,
+                    maxWidth: maxWidth,
+                    maxHeight: maxHeight,
+                    width: width,
+                    height: height,
+                    fitWidthProportion: fitWidthProportion,
+                    fitHeightProportion: fitHeightProportion
+                });
+            }
+
+            return sizeProperties;
+        }
+
+        function getProperties(elements, propertyName) {
+            var properties = [];
+
+            for (var i = 0; i < elements.length; ++i) {
+                properties.push(getProperty(elements[i], propertyName));
+            }
+
+            return properties;
+        }
+
+        // When reading an element's width/height, minWidth/minHeight etc, we have to look in
+        // a few different places in order. This is because the presence of a LayoutChildComponent
+        // on each element is optional, and each property value also has a set of fallback defaults
+        // to be used in cases where no value is specified.
+        function getProperty(element, propertyName) {
+            var layoutChildComponent = element.entity['layoutchild'];
+
+            // First attempt to get the value from the element's LayoutChildComponent, if present.
+            // TODO Should this check if the LayoutChildComponent is enabled?
```

Comment:
@daredevildave what do you think here? Makes sense to check it IMO – i.e. if the `LayoutChildComponent` is disabled then it should have no effect on its element's sizing within the `LayoutGroup`.

Reply:
Yeah, that sounds reasonable